### PR TITLE
MDS-3717: Create Union Rep Party Type + Bug Fixes

### PR DESCRIPTION
**Pull request by exzizt**
# Main
* Users can create a new contact type "Union Rep" on mines
image
image

# Other
* Fixes a bug where contacts could not be updated due to the "Contacts must have a related permit or mine" error
  * May fix [MDS-3550](https://bcmines.atlassian.net/browse/MDS-3550?atlOrigin=eyJpIjoiM2M3MWQ1ZDI2NjM3NGUzYzkzZGFiOTliOWFkMWU2ZmEiLCJwIjoiaiJ9)
* Fixes a bug where permit-linked contacts (Site Operator, Tenure Holder, Land Owner) were not being displayed on the mine Contacts page

### CODE REVIEW

File: services/core-api/app/api/parties/party_appt/models/mine_party_appt.py [hunk 1]
```diff
@@ -183,16 +166,16 @@ def find_by(cls,
...
-        if include_permittees and mine_guid:
+        if include_permit_contacts and mine_guid:
             #avoid circular imports.
             from app.api.mines.mine.models.mine import Mine
             mine = Mine.find_by_mine_guid(mine_guid)
             permit_permittees = []
             for mp in mine.mine_permit:
                 if not active_only:
-                    permit_permittees = permit_permittees + mp.permittee_appointments
+                    permit_permittees = permit_permittees + mp.permit_appointments
```
Comment (exzizt) [inline services/core-api/app/api/parties/party_appt/models/mine_party_appt.py]: Changing these two references of "permittee_appointments" to "permit_appointments" fixes the bug where the other permit-linked contacts were not being displayed on the Contacts page.

File: services/core-api/app/api/parties/party_appt/models/mine_party_appt.py [hunk 2]
```diff
@@ -30,54 +30,38 @@ class MinePartyAppointment(SoftDeleteMixin, AuditMixin, Base):
...
+            self.permit = None
 
         if self.mine_party_appt_type_code in PERMIT_LINKED_CONTACT_TYPES:
             permit = Permit.find_by_permit_guid(related_guid)
-            if not permit:
-                raise AssertionError(f'Permit with guid {related_guid} not found')
-            self.permit_id = permit.permit_id
-        return
-
-    def save(self, commit=True):
```
Comment (exzizt) [inline services/core-api/app/api/parties/party_appt/models/mine_party_appt.py:65]: A lot of this validation was tidied up and moved into the individual `@validates` methods below.

**Review by mmumby — DISMISSED** [review]
🔥

**Review by JustinMacaulay — DISMISSED** [review]
All looks good

File: services/core-api/app/commands.py [hunk 3]
```diff
@@ -63,7 +63,7 @@ def _create_data(num):
                 mine_manager = MinePartyAppointmentFactory(
                     mine=mine, mine_party_appt_type_code='MMG')
                 if len(mine.mine_permit) > 0:
-                    permitee = MinePartyAppointmentFactory(permittee=True, party__company=True)
+                    permittee = MinePartyAppointmentFactory(permittee=True, party__company=True)
```
Comment (jon-funk) [inline services/core-api/app/commands.py:66]: t^2

**Review by jon-funk — APPROVED** [review]
Awesome work

**Pull request merged**